### PR TITLE
Add docs

### LINKED ISSUE #12: Add documentation
The README right now is quite sparse! :)

Need to add full documentation before anyone will be able to properly use it.

### PR DESCRIPTION
**Pull request by merelinguist**
Adds some basic docs.

Closes #12

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -27,39 +33,188 @@
...
 - 🍱 Reliable serialization and deserialization
 - 🔐 Type safety with autocompletion
 - 🐾 Negligible runtime footprint
 - 💫 Framework agnostic
 
-## Example
+
+## Backstory
+
+At [Blitz](https://github.com/blitz-js/blitz), we have struggled with the limitations of JSON. Modern databases often store complex types, like Date objects, but it impossible to easily `stringify` your responses without converting any object types to strings. Moreover, you often have to keep track of which fields you need to convert back to objects in your client!
```
Comment (Skn0tt) [inline README.md:46]:
```suggestion
At [Blitz](https://github.com/blitz-js/blitz), we have struggled with the limitations of JSON. We often find ourselves working with `Date`, `Map`, `Set` or `BigInt`, `JSON.stringify` doesn't support any of them without going through the hassle of manually converting!
```

File: README.md [hunk 2]
```diff
@@ -27,39 +33,188 @@
...
+
+Superjson solves these issues by providing a thin wrapper over `JSON.stringify` and `JSON.parse`. Look at the difference between manually converting invalid fields, and using Superjson to handle this for you:
+
+
+```js
```
Comment (Skn0tt) [inline README.md:51]: Nice idea, but I fear this example is a bit lengthy. Could ellipsize unnecessary lines?

File: README.md [hunk 3]
```diff
@@ -27,39 +33,188 @@
...
+
+Install the library with your package manager of choice, e.g.:
+
+```
+yarn add superjson
+```
+
+## Usage
+
+The simplest way of using `superjson` is with its `stringify` and `parse` functions. If you know how to use `JSON.stringify` in JavaScript, you already know Superjson!
```
Comment (Skn0tt) [inline README.md:153]:
```suggestion
The easiest way to use `superjson` are the `stringify` and `parse` functions. If you know how to use `JSON.stringify` in JavaScript, you already know Superjson!
```

File: README.md [hunk 4]
```diff
@@ -27,39 +33,188 @@
...
+
+Simply stringify any expression you’d like:
```
Comment (Skn0tt) [inline README.md:155]:
```suggestion
Easily stringify any expression you’d like:
```

File: README.md [hunk 5]
```diff
@@ -27,39 +33,188 @@
...
+
+```js
+import { stringify } from 'superjson';
```
Comment (Skn0tt) [inline README.md:158]:
```suggestion
import SuperJSON from 'superjson';
```

Maybe do this and then `SuperJSON.stringify` to make the similarities to `JSON.stringify` obvious

File: README.md [hunk 6]
```diff
@@ -27,39 +33,188 @@
...
+
+const json = stringify({date: new Date(0)});
+```
+
+And parse your JSON like so:
+
+```js
+import { parse } from 'superjson';
+
+const object = stringify(json);
```
Comment (Skn0tt) [inline README.md:168]:
```suggestion
const object = parse(json);
```

File: README.md [hunk 7]
```diff
@@ -27,39 +33,188 @@
...
+```
+
+Alternatively, you can use our lower-level `serializer` and `deserializer` functions. These transform any JavaScript expression into an object which is valid for JSON serialization.
```
Comment (Skn0tt) [inline README.md:171]:
```suggestion
Alternatively, transform any JavaScript value into a JSON-compatible one by using our lower-level `serializer` and `deserializer` functions.
```

Comment (Skn0tt) [reply README.md:171]: (I like to put the desired outcome first before the solution, makes it easier to find what you're looking for)

File: README.md [hunk 8]
```diff
@@ -27,39 +33,188 @@
...
+  test: 'regexp',
+};
 */
 ```
 
+## API
+
+Superjson supports many extra types which JSON does not. You can serialize all these:
+
+| type        | supported by standard JSON? |
```
Comment (Skn0tt) [inline README.md:203]: maybe add another `superjson` column where all checkmarks are green?
This also has the potential to include some Easter egg ;D

File: README.md [hunk 9]
```diff
@@ -80,4 +235,11 @@ Thanks goes to these wonderful people ([emoji key](https://allcontributors.org/d
 <!-- prettier-ignore-end -->
 <!-- ALL-CONTRIBUTORS-LIST:END -->
 
-This project follows the [all-contributors](https://github.com/all-contributors/all-contributors) specification. Contributions of any kind welcome!
\ No newline at end of file
+This project follows the [all-contributors](https://github.com/all-contributors/all-contributors) specification. Contributions of any kind welcome!
+
+## Prior art
+
+There are many libraries similar to Superjson:
```
Comment (Skn0tt) [inline README.md:242]:
```suggestion
Other libraries that aim to solve a similar problem:
```

File: README.md (hunk 5 above)
Comment (Skn0tt) [reply README.md:158]: Tbh, I don't think we need to include the `import` line in the example. That one should be self-explanatory :D

**Comment by merelinguist** [comment]
Okay @Skn0tt ready for another look :)

File: README.md [hunk 10]
```diff
@@ -150,22 +57,20 @@ yarn add superjson
 
 ## Usage
 
-The easiest way to use `superjson` are its `stringify` and `parse` functions. If you know how to use `JSON.stringify`, you already know Superjson!
+The easiest way to use Superjson` is with its `stringify` and `parse` functions. If you know how to use `JSON.stringify`, you already know Superjson!
```
Comment (Skn0tt) [inline README.md:60]:
```suggestion
The easiest way to use Superjson is with its `stringify` and `parse` functions. If you know how to use `JSON.stringify`, you already know Superjson!
```

File: README.md [hunk 11]
```diff
@@ -27,39 +33,93 @@
...
+
+The easiest way to use Superjson` is with its `stringify` and `parse` functions. If you know how to use `JSON.stringify`, you already know Superjson!
+
+Easily stringify any expression you’d like:
 
 ```js
-const input = {
+import superjson from 'superjson';
+
+const json = superjson.stringify({date: new Date(0)});
```
Comment (flybayer) [inline README.md:67]:
```suggestion
const jsonString = superjson.stringify({date: new Date(0)});
```

File: README.md [hunk 12]
```diff
@@ -27,39 +33,93 @@
...
+
+Install the library with your package manager of choice, e.g.:
+
+```
+yarn add superjson
+```
+
+## Usage
+
+The easiest way to use Superjson` is with its `stringify` and `parse` functions. If you know how to use `JSON.stringify`, you already know Superjson!
```
Comment (flybayer) [inline README.md:60]:
```suggestion
The easiest way to use Superjson is with its `stringify` and `parse` functions. If you know how to use `JSON.stringify`, you already know Superjson!
```

**Comment by Skn0tt** [comment]
LGTM! :D

File: README.md [hunk 13]
```diff
@@ -27,39 +33,93 @@
...
-const input = {
+import superjson from 'superjson';
+
+const json = superjson.stringify({date: new Date(0)});
+```
+
+And parse your JSON like so:
+
+```js
+const object = superjson.parse(json);
```
Comment (flybayer) [inline README.md:73]:
```suggestion
const object = superjson.parse(jsonString);
```

**Comment by merelinguist** [comment]
@Skn0tt final check needed.

File: README.md [hunk 14]
```diff
@@ -103,6 +103,72 @@ meta = {
...
+const object = {
+  normal: 'string',
+  timestamp: new Date(),
+  test: /superjson/,
+};
+
+const {json, meta} = serialize(object);
+```
+
+Returns **`json and meta, both JSON values.`**
```
Comment (Skn0tt) [inline README.md:122]:
```suggestion
Returns **`json` and `meta`, both JSON-compatible values.**
```

Comment (Skn0tt) [reply README.md:122]: Could you help me understand why you chose this to be bold? 😊

File: README.md [hunk 15]
```diff
@@ -103,6 +103,72 @@ meta = {
...
+const object = {
+  normal: 'string',
+  timestamp: new Date(),
+  test: /superjson/,
+};
+
+const jsonString = stringify(object);
+```
+
+Returns **`string`**
```
Comment (Skn0tt) [inline README.md:154]:
```suggestion
Returns **`string`**.
```

File: README.md [hunk 16]
```diff
@@ -103,6 +103,72 @@ meta = {
...
+
+#### Examples
+
+```js
+const {json, meta} = serialize(object);
+
+deserialize({json, meta});
+```
+
+Returns **`your original value`**
```
Comment (Skn0tt) [inline README.md:136]:
```suggestion
Returns **`your original value`**.
```

File: README.md [hunk 17]
```diff
@@ -103,6 +103,72 @@ meta = {
...
+
+#### Examples
+
+```js
+const jsonString = stringify(object);
+
+parse(jsonString);s
+```
+
+Returns **`string`**
```
Comment (Skn0tt) [inline README.md:168]:
```suggestion
Returns **`string`**.
```

File: README.md [hunk 18]
```diff
@@ -103,6 +103,72 @@ meta = {
...
+};
+
+const jsonString = stringify(object);
+```
+
+Returns **`string`**
+
+### parse
+
+Parses and then deserializes your the JSON string of `stringify`.
```
Comment (Skn0tt) [inline README.md:158]:
```suggestion
Parses and then deserializes the JSON string returned by `stringify`.
```

File: README.md [hunk 19]
```diff
@@ -103,6 +103,72 @@ meta = {
...
+
+```js
+const jsonString = stringify(object);
+
+parse(jsonString);s
+```
+
+Returns **`string`**
+
+***
```
Comment (Skn0tt) [inline README.md:170]: What does this do? 😅

File: README.md [hunk 20]
```diff
@@ -27,39 +33,159 @@
...
+const jsonString = superjson.stringify({date: new Date(0)});
+```
+
+And parse your JSON like so:
+
+```js
+const object = superjson.parse(jsonString);
+```
+
+Alternatively, transform any JavaScript value into a JSON-compatible one by using our lower-level `serializer` and `deserializer` functions.
```
Comment (robertgrzonka) [inline README.md:76]: I think this should be `serialize` and `deserialize` cause these are the proper names for this functions.

Comment (flybayer) [inline README.md:76]:
```suggestion
Alternatively, transform any JavaScript value into a JSON-compatible one by using our lower-level `serialize` and `deserialize` functions.
```

**Pull request merged**